website/pages/crouch-cancel-calculator/[characterId]/[characterName]/index.tsx

```diff
@@ -53,18 +55,88 @@ export const getStaticProps = (async (context) => {
 }>;
 
 export default function CrouchCancelCalculatorCharacterPage({ data }: InferGetStaticPropsType<typeof getStaticProps>) {
+  const [selectedCharacter, setSelectedCharacter] = useState<string>(data.character.normalizedName.replace('%26', '&'));
+  const [characterData, setCharacterData] = useState<Character | null>(data.character);
+  const [selectedTarget, setSelectedTarget] = useState<number | null>(null);
   const [target, setTarget] = useState<CharacterBase | null>(null);
   const [mode, setMode] = useState<Knockback>(120);
+  const [staleness, setStaleness] = useState<number>(0);
 
-  function setSelectedValue(keys: Set<Key> | 'all') {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const queryParamTarget = searchParams.get('target');
+    if (queryParamTarget) {
+      if (target) {
+        return;
+      }
+      const character = characters.find((character) => character.normalizedName === queryParamTarget);
+      if (!character) {
+        return;
+      }
+      setSelectedTarget(character!.fightCoreId);
+      setTarget(character!);
+    }
+  }, [searchParams]);
+
+  useEffect(() => {
+    setCharacterData(null);
+    fetch(`/framedata/${selectedCharacter}.json`)
```

Comment:
Ideally we don't use fetch but rather something that could actually cache the data client side, is a nice to have to implement later.